cypress/e2e/open.spec.js

```diff
@@ -0,0 +1,122 @@
+			randUser = user
+			cy.login(user)
+			cy.uploadFile(user, 'document.odt', 'application/vnd.oasis.opendocument.text', '/document.odt')
+			cy.uploadFile(user, 'spreadsheet.ods', 'application/vnd.oasis.opendocument.spreadsheet', '/spreadsheet.ods')
+			cy.uploadFile(user, 'presentation.odp', 'application/vnd.oasis.opendocument.presentation', '/presentation.odp')
+			cy.uploadFile(user, 'drawing.odg', 'application/vnd.oasis.opendocument.drawing', '/drawing.odg')
+		})
+	})
+	beforeEach(function() {
+		cy.login(randUser)
+	})
+
+	const fileTests = ['document.odt', 'presentation.odp', 'spreadsheet.ods', 'drawing.odg']
+	fileTests.forEach((filename) => {
+
+		it('Classic UI: Open ' + filename + ' the viewer on file click', function() {
+			cy.nextcloudTestingAppConfigSet('richdocuments', 'uiDefaults-UIMode', 'classic')
+			cy.login(randUser)
+
+			cy.visit('/apps/files', {
+				onBeforeLoad(win) {
+					cy.spy(win, 'postMessage').as('postMessage')
+				},
+			})
+			cy.openFile(filename)
+
+			cy.get('#viewer', { timeout: 15000 })
+				.should('be.visible')
+				.and('have.class', 'modal-mask')
+				.and('not.have.class', 'icon-loading')
+
+			cy.get('#collaboraframe').iframe().should('exist').as('collaboraframe')
+			cy.get('@collaboraframe').within(() => {
+				cy.get('#loleafletframe').iframe().should('exist').as('loleafletframe')
+			})
+
+			cy.get('@loleafletframe').find('#main-document-content').should('exist')
+
+			// FIXME: wait for collabora to load (sidebar to be hidden)
+
+			cy.screenshot('open-file_' + filename)
+
+			// Share action
+			cy.get('@loleafletframe').within(() => {
+				cy.get('#main-menu #menu-file > a').click()
+				cy.get('#main-menu #menu-shareas > a').click()
+			})
+
+			cy.get('#app-sidebar-vue')
+				.should('be.visible')
+			cy.get('.app-sidebar-header__maintitle')
+				.should('be.visible')
+				.should('contain.text', filename)
+			// FIXME: wait for sidebar tab content
+			// FIXME: validate sharing tab
+			cy.screenshot('share-sidebar_' + filename)
+
+			// Validate closing
+			cy.get('@loleafletframe').within(() => {
+				cy.get('#closebutton').click()
+			})
+			cy.get('#viewer', { timeout: 5000 }).should('not.exist')
+		})
+
+		it('Notebookbar UI: Open ' + filename + ' the viewer on file click', function() {
+			cy.nextcloudTestingAppConfigSet('richdocuments', 'uiDefaults-UIMode', 'tabbed')
+			cy.login(randUser)
+
+			cy.visit('/apps/files', {
+				onBeforeLoad(win) {
+					cy.spy(win, 'postMessage').as('postMessage')
+				},
+			})
+			cy.openFile(filename)
+
+			cy.get('#viewer', { timeout: 15000 })
+				.should('be.visible')
+				.and('have.class', 'modal-mask')
+				.and('not.have.class', 'icon-loading')
+
+			cy.get('#collaboraframe').iframe().should('exist').as('collaboraframe')
+			cy.get('@collaboraframe').within(() => {
+				cy.get('#loleafletframe').iframe().should('exist').as('loleafletframe')
+			})
+
+			cy.get('@loleafletframe').find('#main-document-content').should('exist')
```

Comment:
Can we use the waitForCollabora collabora command here?

Reply:
Yep good catch. I replaced all occurrences with the command and cleaned up the unused commands that have been moved to the package already.